### PR TITLE
Derek furst/update derived queries

### PR DESCRIPTION
**Pull request by DerekFurstPitt**
Changed dataset_is_primary to use creation_action to determine whether a dataset is primary rather than inspecting its distance from the nearest sample ancestor

### CODE REVIEW

**Comment by yuanzhou** [comment]
@DerekFurstPitt The query updates handle the "Central Process" and "Lab Process" Activity types. Do we need to exclude the "Multi-Assay Split" type as well for those component datasets?

**Comment by DerekFurstPitt** [comment]
> @DerekFurstPitt The query updates handle the "Central Process" and "Lab Process" Activity types. Do we need to exclude the "Multi-Assay Split" type as well for those component datasets?

I am uncertain. Issue [534](https://github.com/hubmapconsortium/entity-api/issues/534) only says lab process and central process, and the example query bill included specifically uses the "ends with process" bit

**Comment by yuanzhou** [comment]
From Bill:
>The result of a “Multi-Assay Split” dataset isn’t a derived/processed dataset so should not be included in this task.

File: src/app_neo4j_queries.py [hunk 1]
```diff
@@ -626,8 +579,9 @@ def get_prov_info(neo4j_driver, param_dict, published_only):
     if published_only:
         published_only_query_string = f" AND toUpper(ds.status) = 'PUBLISHED'"
         published_only_revisions_string = f" WHERE toUpper(rev.status) = 'PUBLISHED'"
-    query = (f"MATCH (ds:Dataset)<-[:ACTIVITY_OUTPUT]-(a)<-[:ACTIVITY_INPUT]-(firstSample:Sample)<-[*]-(donor:Donor)"
-             f"WHERE not (ds)-[:REVISION_OF]->(:Dataset)"
+    query = (f"MATCH (ds:Dataset)<-[:ACTIVITY_OUTPUT]-(a:Activity)<-[*]-(firstSample:Sample)<-[*]-(donor:Donor)"
+             f" WHERE not (ds)-[:REVISION_OF]->(:Dataset)"
+             f" AND NOT toLower(a.creation_action) ENDS WITH 'process'"
```
Comment (yuanzhou) [inline src/app_neo4j_queries.py:584]: The cypher query Bill used in https://github.com/hubmapconsortium/entity-api/issues/534 only finds the derived/processed datasets. However, you are using negation here, so you may need to double check and exclude the “Multi-Assay Split”.

**Review by yuanzhou — CHANGES_REQUESTED** [review]

File: src/app_neo4j_queries.py (hunk 1 above)
Comment (yuanzhou) [reply src/app_neo4j_queries.py:584]: Confirmed with Bill:
>“Multi-Assay Split” datasets are primary datasets.

**Review by yuanzhou — APPROVED** [review]

**Pull request merged**